h2o-core/src/main/java/water/util/FrameUtils.java

```diff
@@ -60,14 +60,24 @@ public static Frame parseFrame(Key okey, URI ...uris) throws IOException {
   }
 
   public static Key eagerLoadFromHTTP(String path) throws IOException {
-    java.net.URL url = new URL(path);
-    Key destination_key = Key.make(path);
-    java.io.InputStream is = url.openStream();
-    UploadFileVec.ReadPutStats stats = new UploadFileVec.ReadPutStats();
-    UploadFileVec.readPut(destination_key, is, stats);
-    return destination_key;
+    return eagerLoadFromURL(path, new URL(path));
   }
 
+  public static Key<?> eagerLoadFromURL(String sourceId, URL url) throws IOException {
+    InputStream is = url.openStream();
+    return eagerLoadFromInputStream(sourceId, is);
+  }
+
+  public static Key<?> eagerLoadFromInputStream(String sourceId, InputStream is) throws IOException {
+    try {
+      Key<?> destination_key = Key.make(sourceId);
+      UploadFileVec.ReadPutStats stats = new UploadFileVec.ReadPutStats();
+      UploadFileVec.readPut(destination_key, is, stats);
+      return destination_key;
+    } finally {
+      IOUtils.closeQuietly(is);
```

Comment:
I'd rather close this in `eagerLoadFromURL` where the stream is opened.

Reply:
yep, looks better now thanks to try-with-resources